File: Friend.swift
//
//  User.swift
//  Rendezvous
//
//  Created by John Jin Woong Kim on 8/26/15.
//  Copyright (c) 2015 John Jin Woong Kim. All rights reserved.
//

import Foundation

class Friend:NSObject, Comparable{
    var username: String!
    var friendname: String!
    var selected: Bool = false
    var notification:Bool = false
    var time: Date!
    var rendezCount: Int!//simple counters to see notification numbers
    var chatCount: Int!
    var loctime:String!
    var location:String!
    var isGroup:Bool = false
    
    
    init(username: String, showname: String, timestamp: Date){
        self.username = username
        self.friendname = showname
        self.time = timestamp
        self.isGroup = false
    }
    
    init(username: String, showname: String, timestamp: Date,g:Bool){
        self.username = username
        self.friendname = showname
        self.time = timestamp
        self.isGroup = true
    }
    
    init(username: String, showname: String, timestamp: Date, loctime:String, location:String){
        self.username = username
        self.friendname = showname
        self.time = timestamp
        self.loctime = loctime
        self.location = location
        self.isGroup = false
    }
    
    
    func getUsername()-> String! {
    return self.username
    }
    func getShowname()-> String! {
    return self.friendname
    }
    func getTime()->Date{
        return self.time
    }
    
    func isSelected()-> Bool! {
        return selected
    }
}


func <(lhs: Friend, rhs: Friend) -> Bool {
    let notifFlag = lhs.time.compare(rhs.time)
    //print(notifFlag)
    
    if notifFlag == .orderedAscending{
        return true
    }else{
       return false
    }
    
}

func >(lhs: Friend, rhs: Friend) -> Bool {
    let notifFlag = lhs.time.compare(rhs.time)
    //print(notifFlag)
    
    if notifFlag == .orderedDescending{
        return true
    }else{
        return false
    }
    
}

func ==(lhs: Friend, rhs: Friend) -> Bool {
    let notifFlag = lhs.time.compare(rhs.time)
    //print(notifFlag)
    
    if notifFlag == .orderedSame{
        return true
    }else{
        return false
    }
    
}

File: Friend.java
package com.bashlord.loginregister;

/**
 * Created by JJK on 8/10/15.
 */
public class Friend implements Comparable<Friend>{
    String username, friendname, phone, email, location;
    int status, chatCount, rendezCount;
    boolean selected = false;
    boolean notification = false;
    String time;
    java.sql.Timestamp timestamp;//this will be the new loctime
    public Friend(String username, String friendname, String time){
        this.username = username;
        this.friendname = friendname;
        this.time = time;
    }

    public Friend(String username, String friendname, java.sql.Timestamp time){
        this.username = username;
        this.friendname = friendname;
        this.timestamp = time;
    }
    public Friend(String username, String friendname, String phone, String email, int status){
        this.username = username;
        this.friendname = friendname;
        this.phone = phone;
        this.email = email;
        this.status = status;
    }

    public Friend(String username, String friendname, String time, java.sql.Timestamp loctime, String location){
        this.username = username;
        this.friendname = friendname;
        this.time = time;
        this.timestamp = loctime;
        this.location = location;
    }

    public String getUsername() {
        return username;
    }
    public String getShowname() {
        return friendname;
    }


    public boolean isSelected() {
        return selected;
    }
    public void setSelected(boolean selected) {
        this.selected = selected;
    }
    public boolean notifOn() {
        return notification;
    }
    public void notifChange(boolean selected) {
        this.notification = selected;
    }

    @Override
    public int compareTo(Friend another) {
        java.sql.Timestamp currentTime = java.sql.Timestamp.valueOf(this.time);
        java.sql.Timestamp friendTime = java.sql.Timestamp.valueOf(another.time);
        if(currentTime.getTime() < friendTime.getTime()){
            return 0;
        }else return 1;
    }
}
